e2e/env/env.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// This test sets singularity image specific environment variables and
+// verifies that they are properly set.
+
+package singularityenv
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath string `split_words:"true"`
+}
+
+var testenv testingEnv
+
+func singularityEnv(t *testing.T) {
+	// Singularity defines a path by default. See singularityware/singularity/etc/init.
+	var defaultImage = "docker://alpine:3.8"
+	var defaultPath = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+	// This image sets a custom path.
+	var customImage = "docker://godlovedc/lolcow"
+	var customPath = "/usr/games:" + defaultPath
+
+	// Append or prepend this path.
+	var partialPath = "/foo"
+
+	// Overwrite the path with this one.
+	var overwrittenPath = "/usr/bin:/bin"
+
+	var singularityEnvTests = []struct {
+		name  string
+		image string
+		path  string
+		env   []string
+	}{
+		{"DefaultPath", defaultImage, defaultPath, []string{}},
+		{"CustomPath", customImage, customPath, []string{}},
+		{"AppendToDefaultPath", defaultImage, defaultPath + ":" + partialPath, []string{"SINGULARITYENV_APPEND_PATH=/foo"}},
+		{"AppendToCustomPath", customImage, customPath + ":" + partialPath, []string{"SINGULARITYENV_APPEND_PATH=/foo"}},
+		{"PrependToDefaultPath", defaultImage, partialPath + ":" + defaultPath, []string{"SINGULARITYENV_PREPEND_PATH=/foo"}},
+		{"PrependToCustomPath", customImage, partialPath + ":" + customPath, []string{"SINGULARITYENV_PREPEND_PATH=/foo"}},
+		{"OverwriteDefaultPath", defaultImage, overwrittenPath, []string{"SINGULARITYENV_PATH=" + overwrittenPath}},
+		{"OverwriteCustomPath", customImage, overwrittenPath, []string{"SINGULARITYENV_PATH=" + overwrittenPath}},
+	}
+
+	for _, currentTest := range singularityEnvTests {
+		t.Run(currentTest.name, test.WithoutPrivilege(func(t *testing.T) {
+			args := []string{"exec", currentTest.image, "env"}
+
+			cmd := exec.Command(testenv.CmdPath, args...)
+			cmd.Env = append(os.Environ(), currentTest.env...)
+			b, err := cmd.CombinedOutput()
+
+			out := string(b)
+			t.Logf("args: '%v'", strings.Join(args, " "))
```

Comment:
This for sure is a string, so you can use:

```suggestion
			t.Logf("args: '%s'", strings.Join(args, " "))
```